### ISSUE TITLE
issue with russian letters in html-report (utf-8 charset issue)

### ISSUE THREAD

**Issue by 4815162342lost** [darold/pgbadger#554]
Hi,

I have a postgres log file in UTF-8:
```
vodka@vodka-PC:~$ file /tmp/134.log 
/tmp/134.log: UTF-8 Unicode text, with very long lines
```
I have a lot of russian characters in log-file, for example:
```
2020-01-15 13:01:33.711 MSK 392994 vodka@40_vodka from 192.168.0.1 [vxid:12/0 txid:0] [SELECT] LOG:  duration: 1789.424 ms  statement: /*url: /vodka/giveaway/get_totals, usr: anonymous-user, ent: None*/
        --- Запрос дополняющий аналитику текстовой расшифровкой
        WITH data AS (
        SELECT

            z.*,
            -- Остаток на конец = остаток на начало + обороты
            CASE z.activity_type
              WHEN 2 THEN 0
              ELSE z.summa_begin_debet + z.summa_turn_debet - z.summa_turn_credit
            END AS summa_end_debet,

```

I am tried to make html-report via pgbadger:
` ./pgbadger -j 4  -o /tmp/134.html /tmp/134.log  -p '%m %p %u@%d from %h [vxid:%v txid:%x] [%i] '  --charset=utf-8`

But report looks not very well due to unreadable characters:
https://i.imgur.com/KA7ZWOS.png

Could you please take a look?

**Comment by yethee** [comment]
I have same issue with cyrillic characters.

From log file:
```
Jan 21 01:58:41 host123 postgres-main[31995]: [11-1] user=foo-user,db=foo-db,app=[unknown],client=255.255.255.255 LOG:  duration: 489.149 ms  execute <unnamed>: SELECT k0_.keyword AS sclr_0, k0_.impressions AS sclr_1, k0_.updated_at AS sclr_2 FROM keyword_statistics k0_ WHERE k0_.fieldx = $1 k0_.num_words = $2 AND (k0_.tokens @@ to_tsquery($3)) = true ORDER BY k0_.updated_at DESC LIMIT 1
Jan 21 01:58:41 host123 postgres-main[31995]: [11-2] user=foo-user,db=foo-db,app=[unknown],client=255.255.255.255 DETAIL:  parameters: $1 = '?', $2 = '2', $3 = 'камень & почка & !" & ![ & !]'
```
In reports:
![image](https://user-images.githubusercontent.com/559488/72785200-b4aa0600-3c3b-11ea-8dfc-5421e25e0b1d.png)

**Comment by darold** [comment]
Look like using `lc_messages = 'en_US.UTF-8'` could help according to issue #564 . What is the current value in your postgresql.conf?

**Comment by 4815162342lost** [comment]
> Look like using `lc_messages = 'en_US.UTF-8'` could help according to issue #564 . What is the current value in your postgresql.conf?

Hello,

postgres.conf from database-server:
```
-bash-4.2$ cat ./11/postgresql.conf  | grep -i lc
lc_messages = 'en_US.UTF-8'			# locale for system error message
lc_monetary = 'en_US.UTF-8'			# locale for monetary formatting
lc_numeric = 'en_US.UTF-8'			# locale for number formatting
lc_time = 'en_US.UTF-8'				# locale for time formatting
```

We are generating pgbadger report on other server:
```
[root@web-bb-rt-db-136 ~]# env | grep LC
LC_PAPER=ru_RU.UTF-8
LC_ADDRESS=ru_RU.UTF-8
LC_MONETARY=ru_RU.UTF-8
LC_NUMERIC=ru_RU.UTF-8
LC_TELEPHONE=ru_RU.UTF-8
LC_IDENTIFICATION=ru_RU.UTF-8
LC_MEASUREMENT=ru_RU.UTF-8
LC_TIME=ru_RU.UTF-8
LC_NAME=ru_RU.UTF-8 
```

**Comment by darold** [comment]
Ok, I finally found the regression a fix is coming.

**Comment by darold** [comment]
Commit 7aea3bc solves this issue. Can you please run a `make test` and let me know if it reports failure on test 11 and 12 of the 02_basics.t part? It pass on my fr_FR_UTF-8 local and want to know if this is the same at your side.

**Comment by 4815162342lost** [comment]
Thanks for your quick respond!

But sorry, i can be stupid... But how can i run make test?

```
[vodka@vodka-PC pgbadger]# git log -n 1
commit 7aea3bca35afc4b53d579c8f21b3d1df81726d3e
Author: Gilles Darold <gilles.darold@lzlabs.com>
Date:   Tue Feb 18 18:00:47 2020 +0100

    Fix issue #554 about cyrillic and other encoded statement parameters that
    was not reported properly in the HTML report even with custom charset. The
    regression was introduced with a fix to the well known Perl error message
    "Wide charater in print". The patch have been reverted and a new command
    line option --wide-char is available to recover this behavior. Add this
    option to your pgbadger command line if you have message "Wide charater
    in print". Add a regression test with cyrilic and french encoding. Thanks
    to 4815162342lost and yethee for the report.
[root@web-bb-rt-db-136 pgbadger]# ./pgbadger make test
FATAL: logfile "make" must exist!
[vodka@vodka-PC pgbadger]# make test
make: *** No rule to make target `test'.  Stop.
```

Also i found 02_basics.t file, but can not run...
```
[vodka@vodka-PC t]# perl 02_basics.t
Can't locate Test/Simple.pm in @INC (@INC contains: /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at 02_basics.t line 1.
BEGIN failed--compilation aborted at 02_basics.t line 1.
```

Really sorry, but i did have a deal with perl and tests...

**Comment by 4815162342lost** [comment]
Thanks for your quick respond. But i can be stupid and early did not have a deal with perl ans tests... How can i run test? I tried these commands:
```
[vodka@vodka-PC pgbadger]# perl Makefile.PL
Writing Makefile for pgBadger
[vodka@vodka-PC pgbadger]# make
which: no pod2markdown in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/secretnet/bin:/root/bin)
cp pgbadger blib/script/pgbadger
/usr/bin/perl -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/pgbadger
echo "=head1 SYNOPSIS" > doc/synopsis.pod
./pgbadger --help >> doc/synopsis.pod
echo "=head1 DESCRIPTION" >> doc/synopsis.pod
sed -i.bak 's/ +$//g' doc/synopsis.pod
rm doc/synopsis.pod.bak
sed -i.bak '/^=head1 SYNOPSIS/,/^=head1 DESCRIPTION/d' doc/pgBadger.pod
sed -i.bak '4r doc/synopsis.pod' doc/pgBadger.pod
rm doc/pgBadger.pod.bak
Manifying blib/man1/pgbadger.1p
rm doc/synopsis.pod
[vodka@vodka-PC pgbadger]# make test
which: no pod2markdown in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/secretnet/bin:/root/bin)
PERL_DL_NONLAZY=1 /usr/bin/perl "-MExtUtils::Command::MM" "-e" "test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/01_lint.t ......... Can't locate Test/Simple.pm in @INC (@INC contains: /mnt/pgbadger/pgbadger/blib/lib /mnt/pgbadger/pgbadger/blib/arch /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/01_lint.t line 1.
BEGIN failed--compilation aborted at t/01_lint.t line 1.
t/01_lint.t ......... Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
t/02_basics.t ....... Can't locate Test/Simple.pm in @INC (@INC contains: /mnt/pgbadger/pgbadger/blib/lib /mnt/pgbadger/pgbadger/blib/arch /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/02_basics.t line 1.
BEGIN failed--compilation aborted at t/02_basics.t line 1.
t/02_basics.t ....... Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
t/03_consistency.t .. Can't locate Test/Simple.pm in @INC (@INC contains: /mnt/pgbadger/pgbadger/blib/lib /mnt/pgbadger/pgbadger/blib/arch /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/03_consistency.t line 1.
BEGIN failed--compilation aborted at t/03_consistency.t line 1.
t/03_consistency.t .. Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 

Test Summary Report
-------------------
t/01_lint.t       (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
t/02_basics.t     (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
t/03_consistency.t (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
Files=3, Tests=0,  0 wallclock secs ( 0.01 usr  0.01 sys +  0.01 cusr  0.01 csys =  0.04 CPU)
Result: FAIL
Failed 3/3 test programs. 0/0 subtests failed.
make: *** [test_dynamic] Error 2
[vodka@vodka-PC pgbadger]# prove -b t/
t/01_lint.t ......... Can't locate Test/Simple.pm in @INC (@INC contains: /mnt/pgbadger/pgbadger/blib/lib /mnt/pgbadger/pgbadger/blib/arch /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/01_lint.t line 1.
BEGIN failed--compilation aborted at t/01_lint.t line 1.
t/01_lint.t ......... Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
t/02_basics.t ....... Can't locate Test/Simple.pm in @INC (@INC contains: /mnt/pgbadger/pgbadger/blib/lib /mnt/pgbadger/pgbadger/blib/arch /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/02_basics.t line 1.
BEGIN failed--compilation aborted at t/02_basics.t line 1.
t/02_basics.t ....... Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
t/03_consistency.t .. Can't locate Test/Simple.pm in @INC (@INC contains: /mnt/pgbadger/pgbadger/blib/lib /mnt/pgbadger/pgbadger/blib/arch /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/03_consistency.t line 1.
BEGIN failed--compilation aborted at t/03_consistency.t line 1.
t/03_consistency.t .. Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 

Test Summary Report
-------------------
t/01_lint.t       (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
t/02_basics.t     (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
t/03_consistency.t (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
Files=3, Tests=0,  0 wallclock secs ( 0.02 usr  0.00 sys +  0.00 cusr  0.01 csys =  0.03 CPU)
Result: FAIL
```

**Comment by 4815162342lost** [comment]
All tests passed successfully:
```
[vodka@vodka-PC pgbadger]# make test
which: no pod2markdown in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/secretnet/bin:/root/bin)
PERL_DL_NONLAZY=1 /usr/bin/perl "-MExtUtils::Command::MM" "-e" "test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/01_lint.t ......... ok   
t/02_basics.t ....... ok     
t/03_consistency.t .. ok   
All tests successful.
Files=3, Tests=23, 38 wallclock secs ( 0.03 usr  0.01 sys + 13.62 cusr  0.68 csys = 14.34 CPU)
Result: PASS

```

**Comment by darold** [comment]
Ok good news, thanks. Can you confirm the fix?

**Comment by yethee** [comment]
All tests passed:

```
$ perl Makefile.PL && make test
Generating a Unix-style Makefile
Writing Makefile for pgBadger
Writing MYMETA.yml and MYMETA.json
cp pgbadger blib/script/pgbadger
"/usr/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/pgbadger
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/01_lint.t ......... ok
t/02_basics.t ....... ok
t/03_consistency.t .. ok
All tests successful.
Files=3, Tests=23, 40 wallclock secs ( 0.03 usr  0.00 sys + 15.61 cusr  0.53 csys = 16.17 CPU)
Result: PASS
```

I can confirm that this issue fixed by commit 7aea3bc.

```
$ pgbadger -q -I --retention 4 --ident 'postgres-main' --prefix 'user=%u,db=%d,app=%a,client=%h ' --exclude-query '^(VACUUM|COMMIT)' -O '/var/www/pg_reports' ~/postgresql.log
```
![image](https://user-images.githubusercontent.com/559488/74819171-09ab6b80-5311-11ea-8bcb-bc32ae8d8b45.png)

**Comment by darold** [comment]
Good news, thanks for your feedbacks. I'm closing the issue.
